RogueRobotics2987/Season2024
# 4note autos

Pretty nasty PR - I can only speak to what I helped Callie and Mikaela on.

Once this is merged to main, everything will be in main. A feature branch created off main once this is merged in will have everything. Amp April Tag Auto is not going to be faster than Ian's manual practice. even with the ideas I've suggested to improve the april tag. It fell in priority to Auto, anyway. I'll have Callie clean up any April Tag Amp code that's residual, if there is any. 


(some of the) Things this PR covers:
- Removed unneeded Auto paths
- "Just Shoot Center" is now a test path for testing the initial heading, drive direction, and heading reset from auto to teleop. It shots backs up, and that's about it.
- Switched the amp controls from driver to aux controller. Driver still shoots for the amp, however.
- Zero heading at the START of auto, with the robot being rotated 180 (facing alliance wall) - This should work correctly. Removed the "2 *" thing for the offset. it's not needed.
- Started RedFarSideMid path, DO NOT RUN IT - SEE NOTES IN CODE. IT NEEDS WAYPOINTS TO BE UPDATED
- Removed "amparm" from smart dashboard


Things that still need to be done:
-Autopaths should have more consistent, better names. See code for possible names.
-RedFarSideMid and RedSide4Note path needs its waypoint values updated.
-Blue and Red Amp side auto path needs to be created.
-Blue side 4 note and blueFarSideMid should be tested and determined if they need to be honed in (the driving path - not the shooting - that's kinda a mess)
- Remove unneeded poses from GetAutonomousCommand in Robot Container.
- Remove dead code from Robot Container.
- Add Zero Heading to start of all autos - see "JustShootingCenter" as an example. 
(Then all auto paths need to be tested to verify the heading in tele is working after auto.

NOTE: The proper way to test this is EXACTLY like the tournament:
1. Set the robot down where you want it
2. Power it on
3. Choose your auto path
4. Run the practice and let it switch to tele
5. test heading to make sure it's correct.

ANYTHING other than that isn't a test and may give you false positives. (For example - if you set the heading correctly and then run auto, of course heading will be correct, etc)
